Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1

```diff
@@ -90,31 +90,365 @@ Describe 'Identity Cli Functional Tests' {
             }
         }
     }
-    
-    Describe 'Get-ClientRegistration' {}
-        
-    Describe 'New-ClientRegistration' {}
-    
-    Describe 'New-ImplicitClientRegistration' {}
-    
-    Describe 'New-HybridClientRegistration' {}
-    
-    Describe 'New-HybridPkceClientRegistration' {}
-    
-    Describe 'Invoke-UpdateClientRegistration' {}
-    
-    Describe 'Invoke-UpdateClientPassword' {}
-    
-    Describe 'Test-IsClientRegistered' {}
-    
+
+    Describe 'Client Flows' {
+        Context 'Happy Client Lifecycle' {
+            It 'Test, create, update, get, recreate, reset, test' {
+                $timeString = (Get-Date).ToString("hhmmss")
+                $testClientId = "CredentialsHappy$timeString"
+                $fabricToken = Get-FabricInstallerAccessToken -identityUrl $identityUrl -secret $installerSecret
+
+                # Test-IsClientRegistered (expect false)
+                $clientExists = Test-IsClientRegistered -identityUrl $identityUrl -clientId $testClientId -accessToken $fabricToken
+
+                $clientExists | Should -Be $false
+
+                # New-ClientCredentialsClientBody (expect object)
+                $newClient = New-ClientCredentialsClientBody `
+                    -clientId $testClientId `
+                    -clientName "Name for $testClientId" `
+                    -allowedScopes @("fabric/identity.manageresources", "fabric/authorization.read", "fabric/authorization.write", "fabric/authorization.manageclients")
+
+                # New-ClientRegistration (expect client secret)
+                $jsonClient = $newClient | ConvertTo-Json
+                $originalSecret = New-ClientRegistration -identityUrl $identityUrl -body $jsonClient -accessToken $fabricToken
```

Comment:
I didn't catch this in the original PR, but would it make sense to do the json conversion in the New-ClientRegistration method so that its transparent to users of the CLI?

Reply:
Yes, but I'll do that as a separate task / PR?